server/src/modules/models/video/controller.js

```diff
@@ -26,8 +26,13 @@ const setupRoutes = (app) => {
 
   app.get(`${BASE_URL}/detail/:id`, async (req, res) => {
     console.log(`GET`, req.params);
-    const student = await getById(req.params.id);
-    res.send(student);
+    const video = await getById(req.params.id);
+    await Video.updateOne(
```

Comment:
Good idea. But it will be better to move this DB code to next `service` layer and create a function `updateViewCount` 

Why?
We can push the info to the queue for further processing like Who viewed it, what is the current viewers etc.

Reply:
@foyzulkarim vai, i've seen you write multiple update functions (`update`,  `updateHistory`). Both of these perform some update on a single document. There will be more update functions in future. Instead of writing multiple update functions, we can write  single `update` to do any update operations.

```
const updateById = async (id, update) => {
  try {
    const updatedDoc = await Video.updateOne(
      { _id: new ObjectId(id) },
      update
    );
    return updatedDoc;
  } catch (error) {
    return error;
  }
};
```
Now if we want to update video history, we'll call `updateById(documentId, { $push: { history }, $set: { ...rest } })`, or  for view increase `updateById(documentId, { $inc: { viewCount: 1 } })`.

Let me know your thought on this.

Reply:
Sorry. I will check and reply accordingly soon.